Source/VCGeneration/Checker.cs

```diff
@@ -153,8 +153,13 @@ private void SetRlimit(uint rlimit)
     /// </summary>
     private void Setup(Program prog, ProverContext ctx, Split split = null)
```

Comment:
I wonder whether this code leads to nondeterministic behavior when you're running multiple splits in parallel and multiple stability iterations. The splits can share the same checker so they might influence each other's random seeds. That might not be a problem since we're doing stability iterations anyways, but not being able to get reproducible output in that case might be concerning to some.

If you want deterministic behavior, update the RandomSeed of the Split on each iteration, and set `SolverOptions.RandomSeed` using that.

Reply:
On further thought, I think that this approach would have the same issue that the comment above does: methods explicitly annotated with `{:random_seed N}` would instead have a random seed that would evolve over time (though in a predictable way). As it's currently written, the non-determinism you mention would only occur in the case where `/vcsStabilityIterations` is specified in which case I think it's at least arguably desirable.

Reply:
I agree with your assessment, but still I believe that any CLI behavior which is significant, such as whether a Split has failures on some iterations, preferably is reproducible. I think you could get that with minor changes.

Reply:
Yeah, I think you're right. I came up with something that should achieve that while having slightly simpler logic overall.